src/utils/db/db.ts

```diff
@@ -698,52 +707,73 @@ export class DB {
+                plugin_id: pluginConfig.plugin_id.toString(),
+            })
+            return {
+                id: new UUIDT().toString(),
+                team_id: pluginConfig.team_id,
+                plugin_id: pluginConfig.plugin_id,
+                plugin_config_id: pluginConfig.id,
+                timestamp: (timestamp || new Date().toISOString()).replace('T', ' ').replace('Z', ''),
+                source,
+                type,
+                message,
+                instance_id: instanceId.toString(),
+            }
+        })
+
+        if (this.kafkaProducer) {
+            for (const parsedEntry of parsedEntries) {
+                try {
+                    await this.kafkaProducer.queueMessage({
+                        topic: KAFKA_PLUGIN_LOG_ENTRIES,
+                        messages: [{ key: parsedEntry.id, value: Buffer.from(JSON.stringify(parsedEntry)) }],
+                    })
+                } catch (e) {
+                    captureException(e)
+                    console.error(parsedEntry)
+                    console.error(e)
+                }
+            }
+            return
         }
 
         try {
-            if (this.kafkaProducer) {
-                await this.kafkaProducer.queueMessage({
-                    topic: KAFKA_PLUGIN_LOG_ENTRIES,
-                    messages: [{ key: entry.id, value: Buffer.from(JSON.stringify(entry)) }],
-                })
-            } else {
-                await this.postgresQuery(
-                    'INSERT INTO posthog_pluginlogentry (id, team_id, plugin_id, plugin_config_id, timestamp, source,type, message, instance_id) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)',
-                    Object.values(entry),
-                    'insertPluginLogEntry'
-                )
+            let values: any[] = []
+            let valuesString = ''
+            for (let i = 0; i < parsedEntries.length; ++i) {
+                const { id, team_id, plugin_id, plugin_config_id, timestamp, source, type, message, instance_id } =
+                    parsedEntries[i]
+
+                // Creates format: ($1, $2, $3, $4, $5, $6, $7, $8, $9), ($10, $11, $12, $13, $14, $15, $16, $17, $18)
+                valuesString += ' ('
+                for (let j = 1; j <= 9; ++j) {
+                    valuesString += `$${9 * i + j}${j === 9 ? '' : ', '}`
+                }
+                valuesString += `)${i === parsedEntries.length - 1 ? '' : ','}`
+
+                values = [
+                    ...values,
+                    ...[id, team_id, plugin_id, plugin_config_id, timestamp, source, type, message, instance_id],
+                ]
             }
+
+            await this.postgresQuery(
+                `INSERT INTO posthog_pluginlogentry (id, team_id, plugin_id, plugin_config_id, timestamp, source,type, message, instance_id) 
+                VALUES ${valuesString}`,
+                values,
+                'insertPluginLogEntries'
+            )
```

Comment:
I'd like to ask you to rearchitect this bit. Let's put Kafka and Postgres on equal buffering footing to make this as straightforward as possible – since `queueMessage` is already buffered within a `KafkaProducerWrapper`, we should have something like `queuePostgresInsert` buffered within a `PostgresWrapper`. This way there is each of Kafka and Postgres only ever has a single simple layer of buffering, and that layer is Kafka-/Postgres-specific.